clarity_epp/qc/illumina.py

```diff
@@ -7,11 +7,11 @@ def set_avg_q30(lims, process_id):
     process = Process(lims, id=process_id)
     artifact = process.analytes()[0][0]
 
-    if all([udf in artifact.udf for udf in ['% Bases >=Q30 R1', 'Yield PF (Gb) R1', '% Bases >=Q30 R2', 'Yield PF (Gb) R2']]):
-        r1_q30 = artifact.udf['% Bases >=Q30 R1']
-        r1_yield = artifact.udf['Yield PF (Gb) R1']
-        r2_q30 = artifact.udf['% Bases >=Q30 R2']
-        r2_yield = artifact.udf['Yield PF (Gb) R2']
+    if all([udf in artifact.container.udf for udf in ['% Bases >=Q30 R1', 'Yield PF (Gb) R1', '% Bases >=Q30 R2', 'Yield PF (Gb) R2']]):
+        r1_q30 = artifact.container.udf['% Bases >=Q30 R1']
+        r1_yield = artifact.container.udf['Yield PF (Gb) R1']
+        r2_q30 = artifact.container.udf['% Bases >=Q30 R2']
+        r2_yield = artifact.container.udf['Yield PF (Gb) R2']
```

Comment:
I am wondering what else is part of the container. Are there other values that should be retrieved from the container? What are the chances only the Q30 was visible, because it is an important variable, and others might be missed?

Reply:
Only these are required to calculate average Q30. Other stats are stored under container which we can always retrieve we needed.